ui/pages/create-account/connect-hardware/index.js

```diff
@@ -308,27 +264,28 @@ class ConnectHardwareForm extends Component {
   }
 
   renderContent() {
-    if (!this.state.accounts.length) {
+    const { device, unlocked, accounts, selectedAccounts, browserSupported } =
+      this.state;
+
+    const deviceSelectedAndUnlocked = device && unlocked;
+    const showAccountList = deviceSelectedAndUnlocked && accounts.length;
+
+    if (!showAccountList) {
       return (
         <SelectHardware
           connectToHardwareWallet={this.connectToHardwareWallet}
-          browserSupported={this.state.browserSupported}
-          ledgerTransportType={this.props.ledgerTransportType}
+          browserSupported={browserSupported}
         />
       );
     }
 
     return (
       <AccountList
         onPathChange={this.onPathChange}
-        selectedPath={this.props.defaultHdPaths[this.state.device]}
-        device={this.state.device}
-        accounts={this.state.accounts}
-        connectedAccounts={this.props.connectedAccounts}
-        selectedAccounts={this.state.selectedAccounts}
+        device={device}
+        accounts={accounts}
+        selectedAccounts={selectedAccounts}
         onAccountChange={this.onAccountChange}
-        chainId={this.props.chainId}
-        rpcPrefs={this.props.rpcPrefs}
```

Comment:
Some props here are removed as they can be easily added via a separate `redux` connection